lib/puppet/file_system/file_impl.rb

```diff
@@ -75,6 +75,14 @@ def read(path)
     path.read
   end
 
+  def read_preserve_line_endings(path)
+    contents = path.read( :mode => 'rb', :encoding => Encoding::UTF_8)
```

Comment:
So thinking on this a little bit more, I think that we should change this method to look like 

```ruby
def read_preserve_line_endings(path)
    File.read(path)
end
```

That way we can reduce the surface area of this change and continue to preserve the behavior on non-Windows machines in the callsites that have changed in 

https://github.com/puppetlabs/puppet/pull/3722/files#diff-286f148216c044806e2a9d4beafe0925R29

https://github.com/puppetlabs/puppet/pull/3722/files#diff-965778ed2c6767ce9918080492dc0b37R101

Then for Windows specifically we could override the behavior to implement the method the way you have done here.  That way we also subvert the Ruby 1.8.7 / TravisCI issue as well (since we don't care about 1.8.7 on Windows)